content/blogs/cbialorucki-tests-2.md

```diff
@@ -0,0 +1,65 @@
+We decided to focus on syncing as many Wine projects as possible to Wine 10.0 and to cleanly document our changes to the Wine code using C preprocessor guards.
+Wine 10.0 is not the latest version of Wine, but using a common version across the board can help simplify future Wine syncs.
+
+One difficulty with using code from the Wine project is that Wine’s headers are not compatible with Microsoft’s headers used in the public Windows SDK.
+For years, our headers were trying to accommodate both, and because of that our headers weren’t compatible with either Microsoft’s headers or Wine’s headers.
+This necessitated several definition hacks for Wine code, our own code, and code we sync with other projects that expect the official Microsoft headers.
+As part of my syncing efforts for the Wine tests, I created a new directory for Wine headers and separated our headers from the Wine headers.
+Moving forward, we will use up-to-date Wine headers for Wine code and our Microsoft-compatible headers for everything else.
+To reinforce this separation of headers, I also rewrote several of our headers that originated from Wine.
+This header separation helps reduce the number of changes to Wine and other code we would otherwise need to apply.
+In addition, this header separation opens up the possibility of distributing an MIT-licensed SDK separate from ReactOS in the future.
+
+Despite the new header initiative, many Wine tests still need several patches to pass and not crash on Windows Server 2003, Windows Vista, Windows 7, Windows 8.1 and even early versions of Windows 10.
+Wine also doesn’t write their tests with MSVC compiler in mind, so I sometimes have to write more patches for MSVC builds.
+After the Wine tests pass on Windows without failures, I then see if the test crashes on ReactOS.
+Additional test failures on ReactOS are a good thing, they show where we can fix bugs; but when a test crashes, we can’t use it to tell if ReactOS is getting better or getting worse.
+After ensuring synced tests don’t crash on ReactOS, I open a pull request that is then reviewed by other core developers and eventually it gets merged.
+
+## What’s next?
+ReactOS Deutschland e.V. pays for my contract and to host our infrastructure, which includes our website, our wiki, our bug tracker, and our test bots.
+To view the results of test bot runs, we developed a tool called Testman that is integrated into our website.
+Testman was also neglected and now has several issues.
+For example, the tool was developed back when we used SVN for version control and the search interface was designed for that.
+Now that we migrated to git, searching is broken on Testman.
+Test results shown in Testman are also sometimes incorrect, especially when a test spawns child processes.
+I plan to fix these issues with Testman during my contract.
+
+Eventually I would like to see our test bots become part of our CI/CD pipeline.
```

Comment:
uh.. they used to be 🙃 that reminds me to look at that colin mention.

Reply:
IIRC it will show tests, but only when launched manually. Also it would always be green, so it's not really useful.